pkg/handler/templates.go

```diff
@@ -140,3 +150,57 @@ func ParseTemplateFilters(c echo.Context) api.TemplateFilterData {
 
 	return filterData
 }
+
+// DeleteTemplate godoc
+// @summary 		Delete a template
+// @ID				deleteTemplate
+// @Description     This enables deleting a specific template.
+// @Tags			templates
+// @Param  			uuid       path    string  true  "Template ID."
+// @Success			204 "Template was successfully deleted"
+// @Failure      	400 {object} ce.ErrorResponse
+// @Failure     	401 {object} ce.ErrorResponse
+// @Failure      	404 {object} ce.ErrorResponse
+// @Failure      	500 {object} ce.ErrorResponse
+// @Router			/templates/{uuid} [delete]
+func (th *TemplateHandler) deleteTemplate(c echo.Context) error {
+	_, orgID := getAccountIdOrgId(c)
+	uuid := c.Param("uuid")
+
+	template, err := th.DaoRegistry.Template.Fetch(orgID, uuid)
+	if err != nil {
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching template", err.Error())
+	}
+	if err := th.DaoRegistry.Template.SoftDelete(orgID, uuid); err != nil {
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error deleting template", err.Error())
+	}
+	enqueueErr := th.enqueueTemplateDeleteEvent(c, orgID, template)
+	if enqueueErr != nil {
+		if err = th.DaoRegistry.Template.ClearDeletedAt(orgID, uuid); err != nil {
+			return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error clearing deleted_at field", err.Error())
+		}
+		return enqueueErr
```

Comment:
enqueueErr would also need to be returned as an error response, ce.NewErrorResponse()